intercom-common/src/methodinfo.rs

```diff
@@ -53,27 +59,91 @@ impl RustArg {
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum Direction { In, Out, Retval }
 
-#[derive(Debug, PartialEq)]
 pub struct ComArg {
-    pub arg : RustArg,
+
+    /// Name of the argument.
+    pub name: Ident,
+
+    /// Rust type of the raw COM argument.
+    pub ty: Type,
+
+    /// Type handler.
+    pub handler: Rc<TypeHandler>,
+
+    /// Argument direction. COM uses OUT params while Rust uses return values.
```

Comment:
Should we talk about the possibility of multiple OUT-parameters?

We can also decide not to support them.

Reply:
We already have this in the form of tuple return values.

```rust
fn foo(..) -> ComResult<(u8, u16, u32)>
```
->
```C++
HRESULT Foo(.., OUT uint8_t*, OUT uint16_t*, OUT uint32_t* );
```